test_gtest/ParenthesesTests.cpp

```diff
@@ -14,3 +14,594 @@ TEST(ParenthesesTests, givenEmptyStringWhenIsBalancedIsCalledThenResultIsTrue)
+{
+    // GIVEN
+    std::string input{"([{}]])]"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString35WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"([{}])]"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString36WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"({[{)}])]"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString37WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"()[{}]])[]]"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString38WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"([{}]])()]}"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString39WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"([{()}]])]{}()["};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString40WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"{{{}}"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString42WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"{(){()})}})"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString43WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"{())}}{}"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString44WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"{()())}}"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString45WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"{(){})}}"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
+TEST(ParenthesesTests, givenString46WhenIsBalancedIsCalledThenResultIsFalse)
+{
+    // GIVEN
+    std::string input{"{())}}"};
+    Parentheses p;
+
+    // WHEN
+     auto result = p.isBalanced(input);
+
+    // THEN
+    ASSERT_FALSE(result);
+}
```

Comment:
Tutaj podobnie nadmiarowe testy usuńcie i możecie to co zostanie pogrupować wrzucając kilka przypadków do jednego testu.